File: 26.remove-duplicates-from-sorted-array.py
#
# @lc app=leetcode id=26 lang=python3
#
# [26] Remove Duplicates from Sorted Array
#

# @lc code=start
class Solution:
    def removeDuplicates(self, nums: list[int]) -> int:
        i, j = 1, 1
        while (j < len(nums)):
            if nums[j] != nums[j-1]:
                nums[i] = nums[j]
                i+=1
            j+=1
        return i

# @lc code=end

File: 26.remove-duplicates-from-sorted-array.go
package leetcode

/*
 * @lc app=leetcode id=26 lang=golang
 *
 * [26] Remove Duplicates from Sorted Array
 */

// @lc code=start
func removeDuplicates(nums []int) int {
	i, j := 1, 1
	for j < len(nums) {
		if nums[j] != nums[j-1] {
			nums[i] = nums[j]
			i++
		}
		j++
	}
	return i
}

// @lc code=end
